packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -364,27 +377,42 @@ export class SDKClient {
             const transactionRecord: TransactionRecord = await resp.getRecord(this.clientMain);
             const cost = transactionRecord.transactionFee.toTinybars().toNumber();
             this.hbarLimiter.addExpense(cost, currentDateNow);
-            this.logger.info(`${requestIdPrefix} Consensus Node ${transactionName} record response: ${resp.transactionId.toString()} ${Status.Success._code}`);
-            this.logger.trace(`${requestIdPrefix} ${resp.transactionId.toString()} ${callerName} transaction cost: ${transactionRecord.transactionFee}`);
+            this.logger.info(`${requestIdPrefix} ${resp.transactionId} ${callerName} ${transactionName} record status: ${Status.Success} (${Status.Success._code}), cost: ${transactionRecord.transactionFee}`);
             this.captureMetrics(
                 SDKClient.transactionMode,
                 transactionName,
                 transactionRecord.receipt.status,
                 cost,
                 callerName);
+
+            this.hbarLimiter.addExpense(cost, currentDateNow);
+
             return transactionRecord;
         }
         catch (e: any) {
             // capture sdk record retrieval errors and shorten familiar stack trace
             const sdkClientError = new SDKClientError(e);
-            this.captureMetrics(
-                SDKClient.transactionMode,
-                transactionName,
-                sdkClientError.status,
-                0,
-                callerName);
+            let transactionFee: number | Hbar = 0;
+            if (sdkClientError.isValidNetworkError()) {
+                // pull transaction record for fee
+                const transctionRecord = await new TransactionRecordQuery()
```

Comment:
Wondering if we should handle exceptions thrown by these calls?

Reply:
There shouldn't be exceptions since `validateReceiptStatus` is false (which i need to update for the instance above).
I can wrap this in a try catch

Reply:
Addressed